pdr_backend/lake/fetch_ohlcv.py

```diff
@@ -55,6 +56,49 @@ def safe_fetch_ohlcv_ccxt(
         return None
 
 
+@enforce_types
```

Comment:
Main comment: overall this looks great! Low complexity, obvious to read. +10 

My other comments are specific things.